libs/hdf-converters/src/msft-secure-score-mapper.ts

```diff
@@ -0,0 +1,203 @@
+import {
+  SecureScore,
+  ControlScore,
+  SecureScoreControlProfile
+} from '@microsoft/microsoft-graph-types';
+import {ExecJSON} from 'inspecjs';
+import {version as HeimdallToolsVersion} from '../package.json';
+import {BaseConverter, ILookupPath, MappedTransform} from './base-converter';
+
+export class MsftSecureScoreMapper extends BaseConverter {
+  withRaw: boolean;
+  profiles: SecureScoreControlProfile[];
+
+  private getProfiles(controlName: string): SecureScoreControlProfile[] {
+    return this.profiles.filter((profile) => profile.id === controlName);
+  }
+
+  mappings: MappedTransform<
+    ExecJSON.Execution & {passthrough: unknown},
+    ILookupPath
+  > = {
+    platform: {
+      name: 'Heimdall Tools',
+      release: HeimdallToolsVersion
+    },
+    version: HeimdallToolsVersion,
+    statistics: {},
+    profiles: [
+      {
+        name: 'Microsoft Secure Score Scan',
+        title: {
+          transformer: (data: SecureScore) =>
+            `Azure Secure Score report - TenantID: ${data.azureTenantId}`
+        },
+        supports: [],
+        attributes: [],
+        groups: [],
+        status: 'loaded',
+        controls: [
+          {
+            path: 'controlScores',
+            id: {
+              transformer: (data: ControlScore) =>
+                `${data.controlCategory}:${data.controlName}`
+            },
+            title: {
+              transformer: (data: ControlScore) => {
+                const titles = this.getProfiles(data.controlName || '')
```

Comment:
optional: one thing we could do to not have to repeatedly call getProfiles is make it so that controls uses a transformer that would act as a closure.  probably would look like the following:

```
blah

makeControls(controlScores: ControlScore[]): MappedTransform<
  ExecJSON.Control[] & ILookupPath,
  ILookupPath
> {
const controlProfiles = this.getProfiles(data.controlName || '')
return the subsection of the mapping you've already written but now without needing to call getProfiles a bunch of times since now you can just use that variable
}

mapping = {
blah,
profiles: [
{
blah,
controls: {
path: 'controlScores',
transformer: makeControls or something like that
}
}
]
}
```

Haven't counted how many times we do the search.  Might not be necessary if we don't do it that often / there's not that much data to search through so there won't be a significant speed savings.

Reply:
agreed.  the current repeated lookup approach is based off of recommendations from initial design review.  as the data is small, postpone this change pending performance analysis.

Reply:
Decided to go the middle road and just memoize that function instead.